packages/components/src/higher-order/with-focus-outside/index.tsx

```diff
@@ -9,11 +7,21 @@ import {
 	__experimentalUseFocusOutside as useFocusOutside,
 } from '@wordpress/compose';
 
+/**
+ * External dependencies
+ */
+import type { FocusEvent } from 'react';
```

Comment:
Nit: can we have the external dependencies imports _before_ the Wordpress dependencies?

Also, I personally prefer to leave the `React` namespace for this kind of types, since it makes it clear that we're dealing with react events (and not the vanilla `FocusEvent`, like the one referenced in the tests) — something like 

```diff
diff --git a/packages/components/src/higher-order/with-focus-outside/index.tsx b/packages/components/src/higher-order/with-focus-outside/index.tsx
index 14f9ab28c6..d340874b8b 100644
--- a/packages/components/src/higher-order/with-focus-outside/index.tsx
+++ b/packages/components/src/higher-order/with-focus-outside/index.tsx
@@ -7,19 +7,14 @@ import {
 	__experimentalUseFocusOutside as useFocusOutside,
 } from '@wordpress/compose';
 
-/**
- * External dependencies
- */
-import type { FocusEvent } from 'react';
-
 export default createHigherOrderComponent(
 	( WrappedComponent ) => ( props ) => {
 		const [ handleFocusOutside, setHandleFocusOutside ] = useState<
-			undefined | ( ( event: FocusEvent ) => void )
+			undefined | ( ( event: React.FocusEvent ) => void )
 		>( undefined );
 
 		const bindFocusOutsideHandler = useCallback<
-			( node: FocusEvent ) => void
+			( node: React.FocusEvent ) => void
 		>(
 			( node: any ) =>
 				setHandleFocusOutside( () =>

```

Reply:
Done.

Reply:
Sicne we're at it, in order to avoid the confusion, let's apply the same changes to the hook version:

```diff
diff --git a/packages/compose/src/hooks/use-focus-outside/index.ts b/packages/compose/src/hooks/use-focus-outside/index.ts
index 91ec421cc0..3d85b588a9 100644
--- a/packages/compose/src/hooks/use-focus-outside/index.ts
+++ b/packages/compose/src/hooks/use-focus-outside/index.ts
@@ -1,16 +1,3 @@
-/**
- * External dependencies
- */
-import type {
-	FocusEventHandler,
-	EventHandler,
-	MouseEventHandler,
-	TouchEventHandler,
-	FocusEvent,
-	MouseEvent,
-	TouchEvent,
-} from 'react';
-
 /**
  * WordPress dependencies
  */
@@ -63,12 +50,12 @@ function isFocusNormalizedButton(
 }
 
 type UseFocusOutsideReturn = {
-	onFocus: FocusEventHandler;
-	onMouseDown: MouseEventHandler;
-	onMouseUp: MouseEventHandler;
-	onTouchStart: TouchEventHandler;
-	onTouchEnd: TouchEventHandler;
-	onBlur: FocusEventHandler;
+	onFocus: React.FocusEventHandler;
+	onMouseDown: React.MouseEventHandler;
+	onMouseUp: React.MouseEventHandler;
+	onTouchStart: React.TouchEventHandler;
+	onTouchEnd: React.TouchEventHandler;
+	onBlur: React.FocusEventHandler;
 };
 
 /**
@@ -82,7 +69,7 @@ type UseFocusOutsideReturn = {
  * wrapping element element to capture when focus moves outside that element.
  */
 export default function useFocusOutside(
-	onFocusOutside: ( ( event: FocusEvent ) => void ) | undefined
+	onFocusOutside: ( ( event: React.FocusEvent ) => void ) | undefined
 ): UseFocusOutsideReturn {
 	const currentOnFocusOutside = useRef( onFocusOutside );
 	useEffect( () => {
@@ -122,17 +109,18 @@ export default function useFocusOutside(
 	 * @param event
 	 * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#Clicking_and_focus
 	 */
-	const normalizeButtonFocus: EventHandler< MouseEvent | TouchEvent > =
-		useCallback( ( event ) => {
-			const { type, target } = event;
-			const isInteractionEnd = [ 'mouseup', 'touchend' ].includes( type );
-
-			if ( isInteractionEnd ) {
-				preventBlurCheck.current = false;
-			} else if ( isFocusNormalizedButton( target ) ) {
-				preventBlurCheck.current = true;
-			}
-		}, [] );
+	const normalizeButtonFocus: React.EventHandler<
+		React.MouseEvent | React.TouchEvent
+	> = useCallback( ( event ) => {
+		const { type, target } = event;
+		const isInteractionEnd = [ 'mouseup', 'touchend' ].includes( type );
+
+		if ( isInteractionEnd ) {
+			preventBlurCheck.current = false;
+		} else if ( isFocusNormalizedButton( target ) ) {
+			preventBlurCheck.current = true;
+		}
+	}, [] );
 
 	/**
 	 * A callback triggered when a blur event occurs on the element the handler
@@ -141,7 +129,7 @@ export default function useFocusOutside(
 	 * Calls the `onFocusOutside` callback in an immediate timeout if focus has
 	 * move outside the bound element and is still within the document.
 	 */
-	const queueBlurCheck: FocusEventHandler = useCallback( ( event ) => {
+	const queueBlurCheck: React.FocusEventHandler = useCallback( ( event ) => {
 		// React does not allow using an event reference asynchronously
 		// due to recycling behavior, except when explicitly persisted.
 		event.persist();

Reply:
Done as well.